catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/ObjectAdapter.java

```diff
@@ -42,19 +48,29 @@ public static ObjectElement marshalFrom(Attribute attribute) throws CatalogTrans
               "Could not transform Metacard to XML.  Could not serialize Attribute Object value.",
               e);
         }
-        ((ObjectElement) element).getValue().add(baos.toByteArray());
+        element.getValue().add(baos.toByteArray());
       }
     }
     return element;
   }
 
   public static Attribute unmarshalFrom(ObjectElement element) {
     AttributeImpl attribute = null;
-    for (Serializable value : element.getValue()) {
+    String elementName = element.getName();
+
+    for (byte[] value : element.getValue()) {
+      Serializable unmarshalledValue = value;
+      try (ObjectInputStream objectInputStream =
+          new ObjectInputStream(new ByteArrayInputStream(value))) {
+        unmarshalledValue = (Serializable) objectInputStream.readObject();
+      } catch (IOException | ClassNotFoundException | RuntimeException e) {
+        LOGGER.debug("Could not unmarshal Metacard object for attribute: {}", elementName, e);
```

Comment:
❓ We want the byte[] to be set as the value of the attribute in the case where something goes amiss here? We don't want it to be left blank/absent?

Reply:
In my mind that made more sense than leaving it absent. If it's some attribute that may be important to someone's security, for example, the presence of the messed up values in the attribute could be a sign that whatever kind of assertions it's trying to make can't be done. Rather than an empty or non-present attribute which may indicate that there are no assertions to be made at all.